### ISSUE TITLE
Homescreen widget to iPhone

### ISSUE THREAD

**Issue by Solhaga10** [Kemmey/Kemmey-TeslaWatch-Public#217]
Homescreen widget to iPhone
any progress?

**Comment by Kemmey** [comment]
Hey @Solhaga10, 

That is something that I will be introducing, yes. But work has not begun yet. I'm still stabilizing the iOS app and re-introducing latent features from the watch version. 

My thinking about the iOS widget is basically like the watch complications - there will be a few different variants - one with overview of the vehicle stats and one with actions. Mind you, that widgets are limited in functionality, and require launching the app when tapped, much like complications on the watch. So the actions will function pretty much like action complications on the watch...

**Comment by Kemmey** [comment]
Hey @Solhaga10, 

In beta build 365 going out soon to TestFlight, there's a few widgets for you to try out :-D 

They're still under development, but are mostly functional at this point. 

Cheers, 

Kim

**Comment by Solhaga10** [comment]
Widget dosent work 🤔

Skickat från Peters iPhone X

**Comment by Kemmey** [comment]
Yeah, I broke it trying to fix something else. New build hopefully going out in a few minutes...

**Comment by Solhaga10** [comment]
💪🏻

Skickat från Peters iPhone X

**Comment by remy911** [comment]
New build dose not work.
Widget is blank/black

**Comment by Kemmey** [comment]
Yeah, I've had reports of that too. I can't reproduce - store build runs fine on all my devices :-(

Could you try turning off vehicle background and try again?

**Comment by Kemmey** [comment]
And did the previous TestFlight builds work? If so, which build number was the last that worked?

**Comment by Solhaga10** [comment]
Mine is working👍🏻


Skickat från Peters iPhone X

**Comment by remy911** [comment]
I habe no background image

**Comment by remy911** [comment]
none works, all black

**Comment by Kemmey** [comment]
Which phones do you have - both @Solhaga10 and @remy911 

And iOS versions. 

I've successfully tested on 12 pro max and 2018 iPad Pro 12.9" on iOS 14.2

**Comment by Solhaga10** [comment]
Updates in testflight?

Skickat från Peters iPhone X

**Comment by Kemmey** [comment]
And all other devices in simulator - but that might not be representative of actual device as widgets have stricter memory requirements on hardware than simulator :-(

**Comment by remy911** [comment]
X and 14.2

**Comment by Solhaga10** [comment]
iPhone 10s ios 14.3 beta 2

Skickat från Peters iPhone X

**Comment by remy911** [comment]
Which build is the first one with widget?

**Comment by Solhaga10** [comment]
Skickat från Peters iPhone X

**Comment by Kemmey** [comment]
@Solhaga10 Attachments do not work via email - have to do it from GitHub app or website. App is brilliant though

**Comment by Kemmey** [comment]
@remy911 Don't recall which one was first - but I would go backwards from latest :-D 

Did you notice the widgets not working before? Have they ever worked?

**Comment by remy911** [comment]
No never

**Comment by Kemmey** [comment]
@remy911 Ok, that's unfortunate. That would have been really valuable information yesterday before release :-D

**Comment by remy911** [comment]
sorry

**Comment by Kemmey** [comment]
@remy911 is this only the watch for Tesla widgets or is it all widgets? I just did a quick google for "iOS blank widgets" and there are numerous results of users having issues with the built in widgets as well.

**Comment by remy911** [comment]
no i have only blank watch for tesla widget.

**Comment by remy911** [comment]
![Uploading BA9A7A40-4F7E-4017-8D13-2A50EF2B8C55.jpeg…]()

**Comment by remy911** [comment]
![ED3C8EC9-B493-4513-939B-C0D186C4E0F2](https://user-images.githubusercontent.com/67857561/100449459-3cf70400-30b4-11eb-9c94-d88446b025c4.jpeg)

**Comment by Kemmey** [comment]
@remy911 Hey, no worries - this is on me!

**Comment by Kemmey** [comment]
@remy911 did you already try rebooting your phone?

**Comment by remy911** [comment]
no I check it

**Comment by remy911** [comment]
yes sorry it works now :-) shame on me!
does it show the moon ikon?

**Comment by Kemmey** [comment]
@remy911 No shame in that - it should just work! But there must be an iOS 14 bug buried in there...

I did see others in forums stating a reboot did not work - so there's already precedence for it not doing anything. But awesome that you're up and running :-D

**Comment by Kemmey** [comment]
@remy911 The widgets currently do not show the same info as the watch complications. I'll work on that for 1.1.1 - as well as some more detailed action complications for the watch where a complication will both be an action but also show status info.

**Comment by Kemmey** [comment]
If you guys have any ideas or requests as to what you'd like on the widgets, let me know! Just remember that they are mostly like complications, so quite limited in refresh time and actions

**Comment by remy911** [comment]
I realy like it, thank you!
I see now battery in % and KM together on the home screen, just great.
now add the Moon and it is for me 100% perfect!!!
